### PR TITLE
FileResponse works with files asynchronously

### PR DESCRIPTION
**Pull request by Tolmachofof**
## What do these changes do?

The web FileResponse works with files asynchronously
## Are there changes in behavior for the user?
No

## Related issue number
#3313

## Checklist

- [x] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names. 
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Comment by asvetlov** [comment]
thanks!

File: aiohttp/web_fileresponse.py [hunk 1]
```diff
@@ -334,8 +337,8 @@ def __init__(self, path: Union[str, pathlib.Path],
             self.headers[hdrs.CONTENT_RANGE] = 'bytes {0}-{1}/{2}'.format(
                 real_start, real_start + count - 1, file_size)
 
-        with filepath.open('rb') as fobj:
+        with (await loop.run_in_executor(None, filepath.open, 'rb')) as fobj:
```
Comment (aamalev) [inline aiohttp/web_fileresponse.py:340]: @Tolmachofof blocking close of the fobj

Comment (Tolmachofof) [reply aiohttp/web_fileresponse.py:340]: @aamalev thanks for your comment!
I think that something like ```_FileContextManager``` would solve this problem. 
Simplified example:

```python
class _FileContextManager:

    def __init__(self, filepath, flags, start):
        self._filepath = filepath
        self._flags = flags
        self._start = start
        self._fobj = None

    async def __aenter__(self):
        loop = asyncio.get_event_loop()
        self._fobj = await loop.run_in_executor(
            None, self._open, self._filepath, self._start
        )
        return self._fobj

    def _open(self, filepath, start):
        fobj = filepath.open(self._flags)
        if start:
            fobj.seek(start)
        return fobj

    async def __aexit__(self, exc_type, exc_val, exc_tb):
        loop = asyncio.get_event_loop()
        await loop.run_in_executor(None, self._fobj.close)
        self._fobj = None
```
If this acceptable i can make a PR

Comment (aamalev) [reply aiohttp/web_fileresponse.py:340]: I think the conversion `with` to `try..finally` is more suitable. `_FileContextManager` need members loop and executor, and this is too much for a one-time code

Comment (Tolmachofof) [reply aiohttp/web_fileresponse.py:340]: You are right. ```try..finally``` statement sounds good. I will make a PR.

**Pull request merged**